### PR TITLE
🧼  NLP task guides

### PR DESCRIPTION
**Pull request by stevhliu**
A clean commit of the NLP task guides #15564 😭

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
LGTM! Feel free to merge once @sgugger approves

File: docs/source/tasks/question_answering.mdx [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+Question answering tasks return an answer given a question. There are two common forms of question answering:
+
+- Extractive: extract the answer from the given context.
+- Abstractive: generate an answer from the context that correctly answers the question.
+
+This guide will show you how to fine-tune [DistilBERT](https://huggingface.co/distilbert-base-uncased) on the [SQuAD](https://huggingface.co/datasets/squad) dataset for extractive question answering.
+
+<Tip>
+
+See the question answering [task page](https://huggingface.co/tasks/token-classification) for more information about other forms of question answering and their associated models, datasets, and metrics.
```
Comment (flozi00) [inline docs/source/tasks/question_answering.mdx:26]: Is that link correct?

File: docs/source/_toctree.yml [hunk 2]
```diff
@@ -152,8 +166,6 @@
       title: CamemBERT
     - local: model_doc/canine
       title: CANINE
-    - local: model_doc/convnext
-      title: ConvNeXT
```
Comment (sgugger) [inline docs/source/_toctree.yml:156]: This should not be removed.

File: docs/source/_toctree.yml [hunk 3]
```diff
@@ -246,10 +258,6 @@
       title: Pegasus
     - local: model_doc/phobert
       title: PhoBERT
-    - local: model_doc/plbart
-      title: PLBart
-    - local: model_doc/poolformer
-      title: PoolFormer
```
Comment (sgugger) [inline docs/source/_toctree.yml:252]: This not either.

File: docs/source/tasks/language_modeling.mdx [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+-->
+
+# Language modeling
+
+Language modeling predicts words in a sentence. There are two forms of language modeling:
+
+- Causal language modeling: predict the next token in a sequence of tokens, and the model can only attend to tokens on the left.
+- Masked language modeling: predict a masked token in a sequence, and the model can attend to tokens bidirectionally.
+
+This guide will show you how to fine-tune [DistilGPT2](https://huggingface.co/distilgpt2) or [DistilRoBERTa](https://huggingface.co/distilroberta-base) on the [r/askscience](https://www.reddit.com/r/askscience/) subset of the [ELI5](https://huggingface.co/datasets/eli5) dataset for causal and masked language modeling respectively.
```
Comment (sgugger) [inline docs/source/tasks/language_modeling.mdx:20]: Maybe mention another tokenizer/architecture can be used the same way somewhere? This feels like it only works for DistilGPT2/DistilRoBERTa

File: docs/source/tasks/language_modeling.mdx [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+```py
+>>> from transformers import DataCollatorForLanguageModeling
+
+>>> tokenizer.pad_token = tokenizer.eos_token
+>>> data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm_probability=0.15)
+```
+
+## Causal language modeling
+
+<Youtube id="Vpjb1lu0MDk"/>
```
Comment (sgugger) [inline docs/source/tasks/language_modeling.mdx:173]: The task video should be linked in the intro IMO. It doesn't show anything about training.

File: docs/source/tasks/language_modeling.mdx [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+
+```py
+>>> from transformers import AutoModelForCausalLM, TrainingArguments, Trainer
+
+>>> model = AutoModelForCausalLM.from_pretrained("distilgpt2")
+```
+
+<Tip>
+
+If you aren't familiar with fine-tuning a model with Trainer, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```
Comment (sgugger) [inline docs/source/tasks/language_modeling.mdx:189]:
```suggestion
If you aren't familiar with fine-tuning a model with the [`Trainer`], take a look at the basic tutorial [here](training#finetune-with-trainer)!
```

File: docs/source/tasks/language_modeling.mdx [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+
+</Tip>
+
+When you create a batch with [`DataCollatorForLanguageModeling`], instruct the data collator to return `tf.Tensor`'s instead of PyTorch tensors:
+
+```py
+>>> from transformers import DataCollatorForLanguageModeling
+
+>>> data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=False, return_tensors="tf")
+```
```
Comment (sgugger) [inline docs/source/tasks/language_modeling.mdx:234]: This should all go in the same snippet as for PyTorch, using the PT/TF split.

File: docs/source/tasks/language_modeling.mdx [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+
+Call [`fit`](https://keras.io/api/models/model_training_apis/#fit-method) to fine-tune the model:
+
+```py
+>>> model.fit(x=tf_train_set, validation_data=tf_test_set, epochs=3)
+```
+
+## Masked language modeling
+
+<Youtube id="mqElG5QJWUg"/>
```
Comment (sgugger) [inline docs/source/tasks/language_modeling.mdx:288]: Same comment as above.

File: docs/source/tasks/language_modeling.mdx [hunk 9]
```diff
@@ -0,0 +1,408 @@
...
+
+```py
+>>> from transformers import AutoModelForMaskedLM
+
+>>> model = AutoModelForMaskedLM.from_pretrained("distilroberta-base")
+```
+
+<Tip>
+
+If you aren't familiar with fine-tuning a model with Trainer, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```
Comment (sgugger) [inline docs/source/tasks/language_modeling.mdx:304]:
```suggestion
If you aren't familiar with fine-tuning a model with the `Trainer`, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```

File: docs/source/tasks/language_modeling.mdx [hunk 10]
```diff
@@ -0,0 +1,408 @@
...
+
+</Tip>
+
+When you create a batch with [`DataCollatorForLanguageModeling`], instruct the data collator to return `tf.Tensor`'s instead of PyTorch tensors:
+
+```py
+>>> from transformers import DataCollatorForLanguageModeling
+
+>>> data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=False, return_tensors="tf")
+```
```
Comment (sgugger) [inline docs/source/tasks/language_modeling.mdx:350]: Should also be in the same snippet as PyTorch, with the PT/TF split.

File: docs/source/tasks/multiple_choice.mdx [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+
+```py
+>>> from transformers import AutoModelForMultipleChoice, TrainingArguments, Trainer
+
+>>> model = AutoModelForMultipleChoice.from_pretrained("bert-base-uncased")
+```
+
+<Tip>
+
+If you aren't familiar with fine-tuning a model with Trainer, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```
Comment (sgugger) [inline docs/source/tasks/multiple_choice.mdx:145]:
```suggestion
If you aren't familiar with fine-tuning a model with the `Trainer`, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```

File: docs/source/tasks/question_answering.mdx (hunk 1 above)
Comment (sgugger) [reply docs/source/tasks/question_answering.mdx:26]: +1, wrong copy-paste here ;-)

File: docs/source/tasks/question_answering.mdx [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+
+```py
+>>> from transformers import AutoModelForQuestionAnswering, TrainingArguments, Trainer
+
+>>> model = AutoModelForQuestionAnswering.from_pretrained("distilbert-base-uncased")
+```
+
+<Tip>
+
+If you aren't familiar with fine-tuning a model with Trainer, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```
Comment (sgugger) [inline docs/source/tasks/question_answering.mdx:155]:
```suggestion
If you aren't familiar with fine-tuning a model with the `Trainer`, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```

File: docs/source/tasks/question_answering.mdx [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+
+</Tip>
+
+When you create a batch with [`DefaultDataCollator`], instruct the data collator to return `tf.Tensor`'s instead of PyTorch tensors:
+
+```py
+>>> from transformers import DefaultDataCollator
+
+>>> data_collator = DefaultDataCollator(return_tensors="tf")
+```
```
Comment (sgugger) [inline docs/source/tasks/question_answering.mdx:204]: Like before, should be in the same snippet as PyTorch.

File: docs/source/tasks/sequence_classification.mdx [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+
+```py
+>>> from transformers import AutoModelForSequenceClassification, TrainingArguments, Trainer
+
+>>> model = AutoModelForSequenceClassification.from_pretrained("distilbert-base-uncased", num_labels=2)
+```
+
+<Tip>
+
+If you aren't familiar with fine-tuning a model with Trainer, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```
Comment (sgugger) [inline docs/source/tasks/sequence_classification.mdx:95]:
```suggestion
If you aren't familiar with fine-tuning a model with the [`Trainer`], take a look at the basic tutorial [here](training#finetune-with-trainer)!
```

File: docs/source/tasks/sequence_classification.mdx [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+
+</Tip>
+
+When you create a batch with [`DataCollatorWithPadding`], instruct the data collator to return `tf.Tensor`'s instead of PyTorch tensors:
+
+```py
+>>> from transformers import DataCollatorWithPadding
+
+>>> data_collator = DataCollatorWithPadding(tokenizer=tokenizer, return_tensors="tf")
+```
```
Comment (sgugger) [inline docs/source/tasks/sequence_classification.mdx:149]: Should be in the same snippet as PyTorch.

File: docs/source/tasks/summarization.mdx [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+
+```py
+>>> from transformers import AutoModelForSeq2SeqLM, Seq2SeqTrainingArguments, Seq2SeqTrainer
+
+>>> model = AutoModelForSeq2SeqLM.from_pretrained("t5-small")
+```
+
+<Tip>
+
+If you aren't familiar with fine-tuning a model with Trainer, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```
Comment (sgugger) [inline docs/source/tasks/summarization.mdx:114]:
```suggestion
If you aren't familiar with fine-tuning a model with the `Trainer`, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```

File: docs/source/tasks/summarization.mdx [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+
+</Tip>
+
+When you create a batch with [`DataCollatorForSeq2Seq`], instruct the data collator to return `tf.Tensor`'s instead of PyTorch tensors:
+
+```py
+>>> from transformers import DataCollatorForSeq2Seq
+
+>>> data_collator = DataCollatorForSeq2Seq(tokenizer=tokenizer, model=model, return_tensors="tf")
+```
```
Comment (sgugger) [inline docs/source/tasks/summarization.mdx:165]: Should be in the same snippet as PyTorch.

File: docs/source/tasks/token_classification.mdx [hunk 18]
```diff
@@ -0,0 +1,268 @@
...
+
+```py
+>>> from transformers import AutoModelForTokenClassification, TrainingArguments, Trainer
+
+>>> model = AutoModelForTokenClassification.from_pretrained("distilbert-base-uncased", num_labels=2)
+```
+
+<Tip>
+
+If you aren't familiar with fine-tuning a model with Trainer, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```
Comment (sgugger) [inline docs/source/tasks/token_classification.mdx:155]:
```suggestion
If you aren't familiar with fine-tuning a model with the `Trainer`, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```

File: docs/source/tasks/token_classification.mdx [hunk 19]
```diff
@@ -0,0 +1,268 @@
...
+
+</Tip>
+
+When you create a batch with [`DataCollatorWithPadding`], instruct the data collator to return `tf.Tensor`'s instead of PyTorch tensors:
+
+```py
+>>> from transformers import DataCollatorForTokenClassification
+
+>>> data_collator = DataCollatorForTokenClassification(tokenizer=tokenizer, return_tensors="tf")
+```
```
Comment (sgugger) [inline docs/source/tasks/token_classification.mdx:204]: Should go in the same snippet as PyTorch.

File: docs/source/tasks/translation.mdx [hunk 20]
```diff
@@ -0,0 +1,221 @@
...
+
+```py
+>>> from transformers import AutoModelForSeq2SeqLM, Seq2SeqTrainingArguments, Seq2SeqTrainer
+
+>>> model = AutoModelForSeq2SeqLM.from_pretrained("t5-small")
+```
+
+<Tip>
+
+If you aren't familiar with fine-tuning a model with Trainer, take a look at the basic tutorial [here](training#finetune-with-trainer)!
```
Comment (sgugger) [inline docs/source/tasks/translation.mdx:116]:
```suggestion
If you aren't familiar with fine-tuning a model with the [`Trainer`], take a look at the basic tutorial [here](training#finetune-with-trainer)!
```

File: docs/source/tasks/translation.mdx [hunk 21]
```diff
@@ -0,0 +1,221 @@
...
+
+</Tip>
+
+When you create a batch with [`DataCollatorForSeq2Seq`], instruct the data collator to return `tf.Tensor`'s instead of PyTorch tensors:
+
+```py
+>>> from transformers import DataCollatorForSeq2Seq
+
+>>> data_collator = DataCollatorForSeq2Seq(tokenizer=tokenizer, model=model, return_tensors="tf")
+```
```
Comment (sgugger) [inline docs/source/tasks/translation.mdx:167]: Should be in the same snippet as PyTorch.

**Review by sgugger — APPROVED** [review]
Thanks for adding all tasks! I left a few more comments.

File: docs/source/tasks/question_answering.mdx (hunk 1 above)
Comment (stevhliu) [reply docs/source/tasks/question_answering.mdx:26]: Oops, thanks for catching this!

File: docs/source/tasks/language_modeling.mdx (hunk 5 above)
Comment (stevhliu) [reply docs/source/tasks/language_modeling.mdx:173]: Linked both task videos in the intro
